packages/test-framework/src/fixture/pagePreset/AddCouponPreset.ts

```diff
@@ -0,0 +1,16 @@
+import { Page } from '@playwright/test';
+
+import { ApiRequestsSender } from './ApiRequestsSender';
+import { CheckoutPagePreset } from './CheckoutPagePreset';
+
+export class AddCouponPreset implements CheckoutPagePreset {
```

Comment:
I noticed that this preset not only adds a coupon but also performs all the actions of the `PaymentStepAsGuestPreset`. This goes against our goal of encouraging developers to use multiple presets in a single test, like this:

```javascript
await checkout.use(new UseAUDPreset('AUD'));
await checkout.use(new PaymentStepAsGuestPreset());
```
In this situation, I think it's better to use `await checkout.applyCoupon('FREESHIP')` because depending on a hard-coded coupon code seems to restrict the use of a preset.

Reply:
@bc-peng Removed the ApplyCouponPreset and added condition to apply coupon in `PaymentStepAsGuestPreset` itself. Cannot use `await checkout.applyCoupon('FREESHIP')` because then we will get the modal popup again we will need to close that and continue the shipping step.

Reply:
Ideally, I think we want a coupon preset like this:
```
await checkout.use(new PaymentStepAsGuestPreset());
await checkout.use(new ApplyCouponPreset('COUPON CODE'));
```
So `PaymentStepAsGuestPreset`'s purpose is clear. Applying a coupon can be combined into a test case.
Your test case is special, the coupon will trigger a modal to impact the shipping selection.
This is not a general case, does it make sense to use playwright actions like `await checkout.applyCoupon('FREESHIP')`(and more actions) to setup the coupon?

Reply:
`await checkout.use(new PaymentStepAsGuestPreset());
await checkout.use(new ApplyCouponPreset('COUPON CODE'));`

Using above preset will open modal popup at the very beginning because shipping details are changed after Payment step is reached, hence not able to test the remove coupon.

And other way is If we write actions to setup the coupon then we won't get the warning while removing coupon becuase the shipping step needs to be updated after applying coupon and hence no warning after removing coupon. 

Moreover, similar to how completing customer, shipping and billing section is part of PaymentStepAsGuest, adding a coupon can also be part of that process as well because there are scenarios where user will apply coupon before reaching Payments step to check for accurate shipping quote.

Reply:
>Using above preset will open modal popup at the very beginning because shipping details are changed after Payment step is reached, hence not able to test the remove coupon.

This coupon triggers an error modal. Building a preset for this coupon doesn’t feel like a reusable solution.

That's why I think we'd prefer playwright actions for this case.

Perhaps we can combine the current two e2e tests into one, which starts at `PaymentStepAsGuestPreset`, then using playwright actions to:
1. apply the coupon;
2. select the free shipping method which is generated by the coupon;
3. come back to the payment step;
4. refresh page;
5. remove coupon and assert the error modal.

>And other way is If we write actions to setup the coupon then we won't get the warning while removing coupon

This can be avoided by configuring the store shipping to ensure `PaymentStepAsGuestPreset` ends up with a paid shipping method.

Reply:
Updated the code to cover both scenarios in single test and to accomodate above suggestion.